space_packet_parser/packets.py

```diff
@@ -2,34 +2,46 @@
 """Packet containers and parsing utilities for space packets."""
 
 from dataclasses import dataclass, field
-from typing import Union, Optional, Protocol, List
+from typing import Any, Optional, Protocol, List
 
 
-@dataclass
-class ParsedDataItem:
-    """Representation of a parsed parameter
+class DataItem:
+    """Mixin class for storing access to the raw value of a parsed data item.
 
-    Parameters
-    ----------
-    name : str
-        Parameter name
-    unit : str
-        Parameter units
-    raw_value : any
-        Raw representation of the parsed value. May be lots of different types but most often an integer
-    derived_value : float or str
-        May be a calibrated value or an enum lookup
-    short_description : str
-        Parameter short description
-    long_description : str
-        Parameter long description
+    We need to override the __new__ method to store the raw value of the data item
+    on immutable built-in types. So this is just a way of allowing us to inject our
+    own attribute into the built-in types.
```

Comment:
This is where I was concerned about complexity being introduced. But looking at the code, it's really not complex. Glad Python has moved away from the advice not to subclass primitives.